core/src/main/java/org/elasticsearch/script/ScriptService.java

```diff
@@ -90,7 +90,7 @@
     public static final Setting<Integer> SCRIPT_MAX_SIZE_IN_BYTES =
         Setting.intSetting("script.max_size_in_bytes", 65535, Property.NodeScope);
     public static final Setting<Integer> SCRIPT_MAX_COMPILATIONS_PER_MINUTE =
-        Setting.intSetting("script.max_compilations_per_minute", 15, 0, Property.Dynamic, Property.NodeScope);
+        Setting.intSetting("script.max_compilations_per_minute", 15, 0, Property.Dynamic, Property.NodeScope, Property.Deprecated);
```

Comment:
I don't feel like we should deprecate this unless there's an actual replacement. It feel strange to say "don't use this setting, but you can't replace it until 6.0".

I think we should port the `max_compilation_rate` changes to 5.6 so that a user can swap the settings from deprecated to non-deprecated *without* upgrading

Reply:
We do this sometimes (e.g., default paths), it's okay.

Reply:
We discussed this and decided documenting it is fine.